### ISSUE TITLE
Expand to the YOURLS_DB_TABLE_LOG table to allow for City tracking

### ISSUE THREAD

**Issue by amirHizkiya** [YOURLS/YOURLS#3227]
### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/YOURLS/.github/blob/main/CODE_OF_CONDUCT.md).

### Submission validity

- [x] This is not a personal support request, that should be posted on [discussions community](https://github.com/YOURLS/YOURLS/discussions).
- [X] I checked [current issues](https://github.com/YOURLS/YOURLS/issues) and this request isn't a duplicate of an existing issue, opened or closed.

### Summary

Let me start off by saying I did find this issue https://github.com/YOURLS/YOURLS/issues/931  
I found the original thread on an archive site and I understand that it was suggested that this was plugin territory (In 2013). 

Now I'm new to YOURLS, but I did some digging into the source code and it looks like this is blocked on the DB level. The only thing that I could think of to do in our plugin was to override **country_code** with a **Zipcode** or City Name using a filter, but country_code is a **char(2)**. We aren't using the referrer, so we might end up using that, but that's more of a hack. 

Using _post_yourls_info_location_ we could query IPs with Maxmind, but IPs being dynamic, this would be less accurate.

I think the best thing to do would be to add an additional field to YOURLS_DB_TABLE_LOG to allow for logging of city names. I belive that everything else would be possible on a plugin level. 

### Motivation

Expanding the YOURLS_DB_TABLE_LOG table with another field will allow us to add a needed information to log clicks on a city level (e.g. zipcode/city name), rather then only country code.

### Describe the solution

Here's where it gets tricky (as always). 

Being new here I'm not sure of the suggested approach. While I belive that adding another field to the YOURLS_DB_TABLE_LOG table would be the easiest to allow for zipcodes or city names. 

What would you guys suggest?

**Comment by dgw** [comment]
> What would you guys suggest?

The [correct approach](https://github.com/YOURLS/YOURLS/wiki/Dont-Hack-Core) would be to implement storing the city information in a separate database table. Data that is only used by a plugin must be managed by that plugin, not hacked into the core DB schema used by YOURLS core.

In combination with a plugin that handles the actual storing and displaying of city info via hooks (feel free to request any that don't yet exist), this is 100% achievable without modifying the core database.

**Comment by ozh** [comment]
What @dgw said.

The ip to location library we're using is by-passable or expandable by design, so you could easily use another IP db. We're using Maxmind's IP to Country and they have [IP to City](https://dev.maxmind.com/geoip/geolite2-free-geolocation-data?lang=en) so I guess this would be easy to implement.

There are probably plugins in https://github.com/YOURLS/awesome-yourls dealing with ip location to serve as inspiration.

**Comment by amirHizkiya** [comment]
> The [correct approach](https://github.com/YOURLS/YOURLS/wiki/Dont-Hack-Core) would be to implement storing the city information in a separate database table...

I was actually thinking of that, and while I agree with the overall approach, DB operations are expensive. I fear that adding an additional DB request _INLINE_ can have unintended consequences. So if I can just slip in using a hook to add the additional data, I think that would be the way to go. 

> The ip to location library we're using is by-passable or expandable by design, so you could easily use another IP db... 

Yeah I've seen a couple that override the current library and I already have the general direction of how everything should be build, I just hit a road block upon seeing the DB.Schema

.
.
.

If nothing else, being new to YOURLS, I just wanted to make sure that I wasn't fundamentally missing something with my approach.

Thanks!
